include/ceed.h

```diff
@@ -292,20 +292,27 @@ CEED_EXTERN int CeedQRFactorization(CeedScalar *mat, CeedScalar *tau, CeedInt m,
 CEED_EXTERN int CeedQFunctionCreateInterior(Ceed ceed, CeedInt vlength,
     int (*f)(void *ctx, CeedInt nq, const CeedScalar *const *u,
              CeedScalar *const *v), const char *focca, CeedQFunction *qf);
+CEED_EXTERN int CeedQFunctionCreateInteriorFromGallery(Ceed ceed,
+    CeedInt vlength, char* spec,
+    CeedQFunction *qf);
+
 CEED_EXTERN int CeedQFunctionAddInput(CeedQFunction qf, const char *fieldname,
                                       CeedInt ncomp, CeedEvalMode emode);
 CEED_EXTERN int CeedQFunctionAddOutput(CeedQFunction qf, const char *fieldname,
                                        CeedInt ncomp, CeedEvalMode emode);
 CEED_EXTERN int CeedQFunctionSetContext(CeedQFunction qf, void *ctx,
                                         size_t ctxsize);
 CEED_EXTERN int CeedQFunctionApply(CeedQFunction qf, CeedInt Q,
-                                   const CeedScalar *const *u,
-                                   CeedScalar *const *v);
+                                   CeedVector* u, CeedVector* v);
 CEED_EXTERN int CeedQFunctionDestroy(CeedQFunction *qf);
 
 CEED_EXTERN int CeedOperatorCreate(Ceed ceed, CeedQFunction qf,
                                    CeedQFunction dqf, CeedQFunction dqfT,
                                    CeedOperator *op);
+
+CEED_EXTERN int CeedOperatorCreateFromGallery(Ceed ceed, char *spec,
```

Comment:
We can't do an OperatorCreateFromGallery..... We would need to do one for every basis/restriction/ect combo...

Reply:
There was a strong wishes for the Gallery being at Operator level, I tested but this doesn't need to be kept.

Reply:
We could imagine the `OperatorGallery` being able to call a subset of the `QFunctionGallery` and wrapping the `QFunction` in an `Operator`.

Reply:
Agreed. I think it is preferable to make a fully formed QFunction rather than a half formed Operator as we will still have to set the Operator fields either way.